llama-index-integrations/llms/llama-index-llms-ibm/llama_index/llms/ibm/base.py

```diff
@@ -58,10 +69,10 @@ class WatsonxLLM(CustomLLM):
     """
 
     model_id: Optional[str] = Field(
-        default=None, description="Type of model to use.", frozen=True
+        default=None, description="Type of model to use.", allow_mutation=False
```

Comment:
allow_mutations is the pydantic v1 way no? For pydantic v2, it should be frozen=True

https://docs.pydantic.dev/latest/concepts/fields/#immutability

Reply:
Fixed